R/dashboard.Rmd

```diff
@@ -1,453 +1,516 @@
+topten <- read.csv(file.path(repository, "data", "secure", "tab.csv")) # Variables need more informative names
+imp <- read.csv(file.path("data", "secure", "country_year_import.csv")) # I don't have access to this file
+exp <- read.csv(file.path("data", "secure", "country_year_export.csv")) # I don't have access to this file
+
+# Connect to local SQL database if requested
+mydb <- NULL
+historicImports <- NULL
+historicExports <- NULL
+historicImportsLatestMonth <- NULL
+historicExportsLatestMonth <- NULL
+principleCOmmodities <- NULL
+if(params$useSQL){
+  mydb <- dbConnect(MySQL(), user='root', password='', dbname='trade', host='localhost')
+}else{
+  
+  # Load a local csv copy of historic trades data
+  historicImports <- read.csv(file.path(repository, "data", "secure", "tradeStats_historic_IMPORTS_14-09-20.csv"))
+  historicExports <- read.csv(file.path(repository, "data", "secure", "tradeStats_historic_EXPORTS_14-09-20.csv"))
+  
+  # COnvert the statistical value column to numeric
+  historicImports$Value <- as.numeric(historicImports$Value)
+  historicExports$Value <- as.numeric(historicExports$Value)
+  
+  # Convert the date column to dates
+  historicImports$Date <- as.Date(historicImports$Date)
+  historicExports$Date <- as.Date(historicExports$Date)
+  
+  # Select the data for latest month
+  importsLatestMonth <- historicImports[is.na(historicImports$Date) == FALSE & 
+                                        historicImports$Date >= params$startDate & historicImports$Date <= params$endDate, ]
+  exportsLatestMonth <- historicExports[is.na(historicExports$Date) == FALSE & 
+                                        historicExports$Date >= params$startDate & historicExports$Date <= params$endDate, ]
+  
+  # Read in the principle commodities table
+  principleCommodities <- read.xlsx(file.path(repository, "data", "open", "OPN_FINAL_ASY_PrincipleCommoditiesClassifications_31-01-20.xlsx"))
+}
+```
+
+Dashboard
+=====================================
+
+Row
+-------------------------------------
+
+### Goods Import/Export Analysis
+
+```{r month value box}
+valueBox(paste0("Current Month: ", format(params$endDate, "%B")),
+         color = "warning")
+```
+
+### Total Current Month Import )
+
+```{r total import value box}
+# Calculate the total import for latest month
+import_cur <- NULL
+if(params$useSQL){
+  rs4 = dbSendQuery(mydb, "SELECT sum(`Stat..Value`) as TotalImport_current from curtrade WHERE curtrade.CP4 = '4000' or curtrade.CP4 = '4071'  or curtrade.CP4 = '7100'")
+    
+  tab4 <- fetch(rs4, n=-1)
+  import_cur <- ceiling(tab4$TotalImport_current/1000000)
+}else{
+  import_cur <- ceiling(sum(importsLatestMonth$Value, na.rm=TRUE)/1000000)
+}
+
+valueBox(import_cur,
+         icon = 'fa-arrow-left',
+         color = "blue")
+```
+
+### Total Current Month Export
+
+```{r total export value box}
+# Calculate the total export for latest month
+export_cur <- NULL

```

Comment:
Using snake_case here but elsewhere using camelCase - either is fine but try and be consistent